### PR TITLE
feat: add a few optimisations

### PR DESCRIPTION
**Pull request by Rqnsom**
- use `fs_extra` instead of handmade `CopyAllDir` function
- run original package on the max number of threads, while use a single-thread test execution for all mutants within all rayon threads (this increases the overall execution time)
- add additional benchmark for `executing_original_package`
- change formula for calculating average mutant execution time
- move manifest rewriting to `mutator-common` crate so that `move-mutotor` can use `mutator-common` without causing circular dependency
- ensure move-mutator doesn't use the original package path

### CODE REVIEW

File: mutator-common/src/tmp_package_dir.rs [hunk 1]
```diff
@@ -70,3 +70,51 @@ pub fn strip_path_prefix<P: AsRef<Path>>(path: P) -> Result<PathBuf> {
...
+        });
+
+    // Switch back to the original dir.
+    std::env::set_current_dir(curdir)?;
+    fs::write(
+        tempdir.join(SourcePackageLayout::Manifest.path()),
+        manifest_string,
+    )?;
+    Ok(())
+}
```
Comment (Rqnsom) [inline mutator-common/src/tmp_package_dir.rs:120]: Just copied here, no change here

File: move-mutation-test/src/mutation_test.rs [hunk 2]
```diff
@@ -83,13 +87,21 @@ pub(crate) fn run_tests_on_mutated_code(
     let mut error_writer = std::io::sink();
 
     // Do not calculate the coverage on mutants.
-    let test_config = cfg.disable_coverage();
+    let mut test_config = cfg.clone();
+    test_config.apply_coverage = false;
+    test_config.ignore_compile_warnings = true;
+    test_config.move_pkg.skip_attribute_checks = true;
```
Comment (Rqnsom) [inline move-mutation-test/src/mutation_test.rs:93]: Less clutter to the stdout while running the tool

File: move-mutation-test/src/lib.rs [hunk 3]
```diff
@@ -165,6 +170,12 @@ pub fn run_mutation_test(
             .into_iter()
             .unzip();
 
+        chunk_iter += 1;
+        info!(
+            "update: finished running tests for {} mutants",
```
Comment (Rqnsom) [inline move-mutation-test/src/lib.rs:175]: A bit more inforamtive output to see the progress

**Review by asmie — APPROVED** [review]
👍 

There is still a problem with short options:
```
RUST_LOG=info ./target/debug/move-mutation-test run -p move-mutator/tests/move-assets/relative_dep/p2/ --verify-mutants
thread 'main' panicked at /home/asmie/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.20/src/builder/debug_asserts.rs:112:17:
Command run: Short option names must be unique for each argument, but '-f' is in use by both 'mutate_functions' and 'filter'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: move-mutator/src/output.rs [hunk 4]
```diff
@@ -121,7 +121,7 @@ pub(crate) fn setup_output_dir(mutator_configuration: &Configuration) -> anyhow:
     let _ = fs::remove_dir_all(&output_dir);
     fs::create_dir(&output_dir)?;
 
-    debug!("Output directory set to: {output_dir:?}");
+    info!("Output directory set to: {output_dir:?}");
```
[the remark this answers is not in the document]
Comment (Rqnsom) [reply move-mutator/src/output.rs:124]: Well, I use that directory quite often so I would say yes 😄

File: move-mutation-test/src/mutation_test.rs [hunk 5]
```diff
@@ -83,13 +87,21 @@ pub(crate) fn run_tests_on_mutated_code(
...
+
+    // Rayon pool will utilize all CPU threads anyway, so one test thread per the bigger rayon
+    // thread should be more than enought. Using more threads here slows the overall time.
```
[the remark this answers is not in the document]
Comment (Rqnsom) [reply move-mutation-test/src/mutation_test.rs]: Done

**Pull request merged**